config/api/src/main/java/org/jboss/windup/config/GraphRewrite.java

```diff
@@ -34,12 +39,35 @@ public GraphRewrite(Iterable<RuleLifecycleListener> listeners, GraphContext cont
         this.graphContext = context;
     }
 
+    /**
+     * @return The exception which holds information where Windup has stopped before finishing (typically on an external request),
+     *      or null if Windup finished normally.
+     */
+    public WindupStopException getWindupStopException()
+    {
+        return windupStopException;
+    }
+
+    /**
+     * Stores the exception which holds information if, and where, the Windup stopped (typically on an external request).
+     * If windup was stopped, this must be called.
+     */
+    public void setWindupStopException(WindupStopException windupStopException)
```

Comment:
I'm still not sure how calling this twice would necessarily be wrong. Couldn't that just mean that the user had clicked "cancel" more than once or something?